Feature/crawling

Jimin0304 opened the pull request.

Jimin0304 committed:
feat(model):add Comment, Post, PostLike, PostScrap, User Entity

Jimin0304 committed:
Chore: DDD 1차 완성

Chat, Follow, HashTag, SearchRecord Entity 완성 후,
Entity의 Repository 추가
User, Post, PostLike, PostScrap, Comment의 @Builder 속 Assert.notNull 추가
에러 핸들링을 위해 기존 Assert.hasText도 Assert.notNull로 수정
forigen key(user_id, post_id)에 @NotNull 추가

Jimin0304 committed:
chore: Entity 수정(#3)

- 폴더명 변경
- 통일성을 위해 @NotNull -> @Column(nullable = false)로 변경

Jimin0304 committed:
feat: twitter 데이터 받아 옴 (#11)

Jimin0304 committed:
Merge branch 'develop' into feature/crawling

# Conflicts:
#	src/main/java/com/capstone/bbiyong/common/domain/DateTimeEntity.java
#	src/main/resources/application-settings.yml
#	src/main/resources/application.yml

Jimin0304 committed:
feat(twitter): twitter api 파싱 및 db에 저장 + scheduler (#11)

- 서울교통공사 twitter를 파싱하여 지하철 지연 정보를 db에 저장
- 1분 간격으로 scheduler를 돌려 새로운 정보가 있으면 받아오게 함

Jimin0304 committed:
feat(accident): accident api scheduler 구현 및 date 형식 변경 (#12)

- 1분 간격으로 scheduler를 돌려 새로운 정보가 있으면 받아오게 함
- 기존 startDate와 endDate가 제대로 받아지지 않아 수정하고, date뿐만 아니라 time도 함꼐 받아오게 함

Jimin0304 committed:
chore(build): 빌드 및 db 설정 수정 (#12)

- 빌드 및 db 설정 수정
- OpenAPI의 parseDateFormat 인자 추가

Jimin0304 committed:
build(gradle.yml):  github의 gradle.yml에 twitter 키 추가 (#11)

Jimin0304 committed:
chore(twitter): twitter 키 직접 주입 (#11)

siyeonSon commented on src/main/java/com/capstone/bbiyong/twitter/domain/Twitter.java:
```diff
@@ -0,0 +1,34 @@
+@Getter
+@NoArgsConstructor
+@Entity
+public class Twitter extends DateTimeEntity {
```
twitter에서 받아오는 데이터가 서울 지하철 정보인데, table(또는 엔티티) 명을 Twitter 대신에 `Metro`라고 하는 게 어떨까요?
결국 실제 서버 관리할 때는 이 데이터가 twitter로부터 왔음이 중요한 게 아니라, 해당 엔티티는 지하철 정보와 관련된 데이터라는 것이 더 중요할 것 같아요.

예를 들어 지하철 정보와 관련해서 급하게 해결해야 하는 이슈가 생겼을 때, Service 로직의 변경이 필요하다 가정해볼게요. _지하철 정보는 트위터로부터 왔으니깐 TwitterService를 수정해야겠다!_ 보다는 _지하철 정보를 담당하는 MetroService를 수정해야겠다!_ 가 조금 더 자연스러운 것 같아서요!!

꼭 Metro가 아니더라도 지민님이 원하는 이름이면 다 좋아요 😀

Jimin0304 replied:
좋은 의견인 것 같아 즉시 반영 후 commit 완료 했습니다 👍🏻

Jimin0304 committed:
chore(entity): twitter entity -> metro entity로 변경 (#11)

- twitter db가 지하철 정보와 관련된 데이터를 받아오는 것을 감안해 twitter보다는 metro가 더 자연스러울 것 같아 entity 이름 변경

Jimin0304 committed:
feat(emerMsg): 재난문자 api 파싱 및 db에 추가 (#13)

Jimin0304 committed:
chore(emerMsg): emergency message key 환경변수 추가 (#13)

Jimin0304 committed:
feat(parse): 데이터 파싱 시 새롭게 업데이트 된 데이터만 db에 저장 (#14)

- 기존에는 전역변수를 사용하여 마지막 데이터를 확인
- Optional을 사용해서 해당되는 id가 없으면 추가하는 방식으로 변경

Jimin0304 committed:
chore(emerMsg): service key 인식 불가 사항 수정 (#14)
